core/src/main/scala/za/co/absa/spline/core/SparkLineageInitializer.scala

```diff
@@ -58,31 +63,59 @@ object SparkLineageInitializer extends Logging {
-          log info s"Successfully initialized. Spark Lineage tracking is ENABLED."
-        } catch {
-          case NonFatal(e) if configurer.splineMode == BEST_EFFORT =>
-            log.error(s"Initialization failed! Spark Lineage tracking is DISABLED.", e)
+      val splineConfiguredForCodelessInit = sparkSession.sparkContext.getConf
+        .getOption(QUERY_EXECUTION_LISTENERS.key).toSeq
+        .flatMap(s => s.split(",").toSeq)
+        .contains(classOf[SplineQueryExecutionListener].getCanonicalName)
+      if (!splineConfiguredForCodelessInit || spark.SPARK_VERSION.startsWith("2.2")) {
+        if (splineConfiguredForCodelessInit) {
+          log.warn("""
+            |Spline lineage tracking is also configured for codeless initialization, but codeless init is
+            |supported on Spark 2.3+ and not current version 2.2. Spline will be initialized only via code call to
+            |enableLineageTracking i.e. the same way as is now."""
+              .stripMargin.replaceAll("\n", " "))
         }
+        createEventHandler(configurer)
+          .foreach(h => sparkSession.listenerManager.register(new SplineQueryExecutionListener(Some(h))))
+      } else {
+        log.warn("""
+          |Spline lineage tracking is also configured for codeless initialization.
+          |It wont be initialized by this code call to enableLineageTracking now."""
+            .stripMargin.replaceAll("\n", " "))
       }
       sparkSession
     }
 
-    /**
-      * The method tries to initialize the library with external settings.
-      *
-      * @param configurer External settings
-      */
-    def attemptInitialization(configurer: SplineConfigurer): Unit = {
-      SparkVersionRequirement.instance.requireSupportedVersion()
-      sparkSession.listenerManager register configurer.queryExecutionListener
+    def createEventHandler(): Option[QueryExecutionEventHandler] = {
+      val configurer = new DefaultSplineConfigurer(defaultSplineConfiguration, sparkSession)
+      if (configurer.splineMode != DISABLED) {
+        createEventHandler(configurer)
+      } else {
+        None
+      }
+    }
 
-      // TODO: SL-128
-//       sparkSession.streams addListener configurer.streamingQueryListener
+    private def createEventHandler(configurer: SplineConfigurer): Option[QueryExecutionEventHandler] = {
+      if (configurer.splineMode != DISABLED) {
+        if (!getOrSetIsInitialized()) {
+          log info s"Spline v${SplineBuildInfo.version} is initializing..."
+          try {
+            SparkVersionRequirement.instance.requireSupportedVersion()
+            val eventHandler = configurer.queryExecutionEventHandler
+            log info s"Spline successfully initialized. Spark Lineage tracking is ENABLED."
+            Some(eventHandler)
+          } catch {
+            case NonFatal(e) if configurer.splineMode == BEST_EFFORT =>
+              log.error(s"Spline initialization failed! Spark Lineage tracking is DISABLED.", e)
```

Comment:
Shouldn't it be warning?

Reply:
In this case I would use error. Initialization failed due to unhandled exception, which we here consume. The Spline code cannot continue, so I would raise it on error log level, such that it is not easily ignored.